scatterer/Utilities/Misc/Utils.cs

```diff
@@ -58,18 +58,12 @@ public static void LogError(string log)
 			Debug.LogError("[Scatterer][Error] " + log);
 		}
 
-		public static GameObject GetMainMenuObject(CelestialBody celestialBody)
+		public static bool GetMainMenuObject(CelestialBody celestialBody, out GameObject gameObject)
 		{
+			// Search all Rotatos for the one with the correct name.
 			string name = celestialBody.isHomeWorld ? "Kerbin" : celestialBody.name;
-
-			GameObject mainMenuObject = GameObject.FindObjectsOfType<GameObject>().FirstOrDefault(b => ( (b.name == name) && b.transform.parent.name.Contains("Scene")));
-			
-			if (ReferenceEquals(mainMenuObject,null))
-			{
-				throw new Exception("No correct main menu object found for "+celestialBody.name);
-			}
-			
-			return mainMenuObject;
+			gameObject = GameObject.FindObjectsOfType<Rotato>().Select(r => r.gameObject).FirstOrDefault(b => b.name == name);

```

Comment:
What exactly is "Rotato"? This doesn't come up in the unity documentation

Reply:
https://www.kerbalspaceprogram.com/ksp/api/class_rotato.html

Rotato is the component KSP uses to make Kerbin spin in the main menu. It should be faster than iterating over each game object in the scene for every celestial body registered with Scatterer, but I suppose it‘s possible that it could miss the planet if there’s any mods that add a planet to the loading screen without making them spin. If you’re worried about that then swapping Rotato for MeshRenderer would be good as well.

Reply:
Thanks for the info. I think it should be fine.